routing/payment_session.go

```diff
@@ -363,3 +366,65 @@ func (p *paymentSession) RequestRoute(maxAmt, feeLimit lnwire.MilliSatoshi,
 		return route, err
 	}
 }
+
+// UpdateAdditionalEdge updates the channel edge policy for a private edge. It
+// validates the message signature and checks it's up to date, then applies the
+// updates to the supplied policy. It returns a boolean to indicate whether
+// there's an error when applying the updates.
+func (p *paymentSession) UpdateAdditionalEdge(msg *lnwire.ChannelUpdate,
+	pubKey *btcec.PublicKey, policy *channeldb.ChannelEdgePolicy) bool {
+
+	// Validate the message signature.
+	if err := VerifyChannelUpdateSignature(msg, pubKey); err != nil {
+		log.Errorf(
+			"Unable to validate channel update signature: %v", err,
+		)
+		return false
+	}
+
+	// Ignore outdated message and return true to indicate that this update
+	// is meant for private edge although we won't apply it.
+	lastUpdate := time.Unix(int64(msg.Timestamp), 0)
+	if !lastUpdate.Before(policy.LastUpdate) {
```

Comment:
Fantastic catch! This is actually related to the `itest` flake, cc @halseth, after tracing down the logs, found this line,
```
2021-05-19 00:26:11.489 [WRN] CRTR: Ignoring outdated update for private edges:500277790703616
```
This means the channel update is not applied, thus causing the error `FAILURE_REASON_NO_ROUTE`.

Reply:
I wanted to validate the timestamp here because of what we did for the normal [channel policy update](https://github.com/lightningnetwork/lnd/blob/master/routing/router.go#L1544). However, since the orders cannot be guaranteed, we could just blindly apply the update once we see one. (Unless the update message includes a timestamp such as a block height, we cannot tell what we received is newer than what we already have.)